addons/product/models/product.py

```diff
@@ -395,17 +395,36 @@ def _name_get(d):
         self.check_access_rule("read")
 
         result = []
+
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        # Use `load=False` to not call `name_get` for the `product_tmpl_id`
+        self.sudo().read(['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids'], load=False)
```

Comment:
@beledouxdenis Are you sure of all these sudoes() ??? Have you tested it in Multi Company environment? That seems weird to me.

Reply:
I just respect what was done before few lines below:
```
 for product in self.sudo():
```

Reply:
Ok, I've just saw that. That's weird too as name_get should display what can be seen by user. The only field read in that  loop is seller_ids which as security group as base.group_user. So, IMHO no need of sudo() at all. But stable policy ?

Reply:
My only concern was the performance. I saw these weird things as well, but if there is something to change, then it's another commit. It was out of context of this commit.

I just did the prefetch thing with the read, and translated the behavior regarding the partners/sellers things, but no more :).

Reply:
Sad than `['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids']` is not provided by an external method. Really useful to override in ecommerce cases.

Reply:
I thought about that as well, but

- I wouldn't do that in a stable version without having a real use case,
- This can be achieved in your override by calling yourself the `read` before calling the super, with your own list of arguments in addition to the ones below. It will fetch the data in the cache, and the second call to `read` will not do any SQL request I believe, as everything will already be in the cache. The only downside is that you have to rewrite the list `['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids']` in your override.

If you explain the use case, and consider my above work-around not decent, I can consider adding a `_prefetch_fields_name_get` method.